packages/angular_devkit/build_webpack/src/utils.ts

```diff
@@ -19,14 +19,17 @@ export function getEmittedFiles(compilation: webpack.compilation.Compilation): E
   const getExtension = (file: string) => file.split('.').reverse()[0];
   const files: EmittedFiles[] = [];
 
-  for (const chunk of Object.values(compilation.chunks)) {
-    const entry: Partial<EmittedFiles> = {
-      name: chunk.name,
-      initial: chunk.isOnlyInitial(),
-    };
-
-    for (const file of chunk.files) {
-      files.push({ ...entry, file, extension: getExtension(file) } as EmittedFiles);
+  for (const key of compilation.entrypoints.keys()) {
```

Comment:
Can you explain a bit why this is needed? This won't return all the files like `compilation.chunks`.

Also, some tests to verify that it generates what you need would be great.

Reply:
It's about determining the right order of the script tags: We just have the right order of the entrypoints. However, one entry point can lead to several files, like `runtime.js` which is not an entrypoint itself. This was not reflected by the original result of this method.

Reply:
Btw: This new logic was original part of the `IndexHtmlPlugin`